idp_node/mapapp/retriever.py

```diff
@@ -0,0 +1,174 @@
+'''
+Build the strings for myproxy-server to use in myproxy-logon,
+See http://grid.ncsa.illinois.edu/myproxy/man/myproxy-server.config.5.html
+'''
+import argparse
+import ConfigParser
+import logging
+import os
+import sys
+
+import OpenSSL
+import psycopg2
+
+def db_connection(config_parser, esgf_home):
+    pass_file = os.path.join(esgf_home, "config", ".esg_pg_pass")
+    logging.info("Retrieving pg pass from %s", pass_file)
+    with open(pass_file, "rb") as filep:
+        password = filep.read().strip()
+
+    section = "installer.properties"
+    db = config_parser.get(section, "db.database")
+    host = config_parser.get(section, "db.host")
+    port = config_parser.get(section, "db.port")
+    user = config_parser.get(section, "db.user")
+
+    connection_params = (
+        "dbname=%s user=%s password=%s host=%s port=%s" %
+        (db, user, password, host, port)
+    )
+    connection_params_no_pass = (
+        "dbname=%s user=%s host=%s port=%s" %
+        (db, user, host, port)
+    )
+    logging.info(connection_params_no_pass)
+
+    return psycopg2.connect(connection_params)
+
+def build_grouprole(args, config_parser, esgf_home):
+    ''' Build the group and role string for a specified username '''
+    logging.info("Fetching groups and roles for %s", args.username)
+
+    section = "installer.properties"
+    esgf_host = config_parser.get(section, "esgf.host")
+    gateway = "https://{}/esgf-idp/openid/%".format(esgf_host)
```

Comment:
Is that trailing percent sign supposed to be standalone or is this a typo?

Reply:
It is the "wildcard" character for the LIKE clause in the SQL statements.